### PR TITLE
Revising tgrains docfix

### PR DESCRIPTION
**Pull request by tiagmoraismorgado**

### CODE REVIEW

File: HelpSource/Classes/TGrains.schelp [hunk 1]
```diff
@@ -3,12 +3,43 @@ summary:: Buffer granulator.
...
 code::
 (sin2(x) for x from 0 to π)
 ::
 and is panned between two channels of multiple outputs.
 
+/////////////////////
+/////////////////////
+/////////////////////
+
+one cool thing you can do, is actually grabbing really large buffers (i.e.: one hour length buffers), and loading them into a buffer, and scrubbing through it in all possible forms. it works particularly well with post-modern kinds of musicm but also with another genres of music, and is a typical kind of scenario you might want to use when you grab a controller such as a wiimote, or a bi-dimensional kind of thing
```
Comment (kisielk) [inline HelpSource/Classes/TGrains.schelp]: There's a typo here: "post-modern kinds of **musicm**"

**Comment by snappizz** [comment]
Something's screwy here -- this PR includes both the edits to SinOsc (PR #2146) and TGrains.

File: HelpSource/Classes/TGrains.schelp (hunk 1 above)
Comment (tiagmoraismorgado) [inline HelpSource/Classes/TGrains.schelp]: how do you suggest me to fix this?

Comment (tiagmoraismorgado) [inline HelpSource/Classes/TGrains.schelp]: can you point a possible solution to handle this?

**Comment by snappizz** [comment]
Instead of doing the fancy cherrypicking magic and separating out the two PRs, let's close the other two and just work on this one. It's generally not best practice to submit two separate changes in a single PR, but here I don't think it's a big deal. It's just helpfiles.

**Comment by snappizz** [comment]
It was really considerate of @bagong to give you a walkthrough of Git 101, but I think the method is both confusing and overkill for helpfile editing. (Sorry Rainer) Instead, I suggest you use the github web interface.

Here are the steps to adding changes to this PR:
- Navigate to [your fork of supercollider](https://github.com/tiagmoraismorgado/supercollider). Under the Branch dropdown, select "revisingTgrainsDocfix".
- Navigate to HelpSource/Classes/ to edit the files.
- Any edits you make will automatically show up here. There's no need to file a new PR for every edit.

Be sure to *only* make edits that are relevant to this PR. So while you're on the revisingTgrainsDocfix branch, be sure only to edit SinOsc.schelp and TGrains.schelp.

**Comment by tiagmoraismorgado** [comment]
agree. Rainer was pretty helpful. i didn't master the whole commands and
concepts behind this thing yet, it will take time. but i will do in mean
time, as i get tougher on this (i am currently on a slow thinking stage :P,
but things will get better soon, i am sure). thank you guys for all your
generosity. and hope hope to honestly keep in touch with the community.
cheers

2016-06-03 18:29 GMT+01:00 Nathan Ho <notifications@github.com>:

> It was really considerate of @bagong <https://github.com/bagong> to give
> you a walkthrough of Git 101, but I think the method is both confusing and
> overkill for helpfile editing. (Sorry Rainer) Instead, I suggest you use
> the github web interface.
>
> Here are the steps to adding changes to this PR:
>
>    - Navigate to your fork of supercollider
>    <https://github.com/tiagmoraismorgado/supercollider>. Under the Branch
>    dropdown, select "revisingTgrainsDocfix".
>    - Navigate to HelpSource/Classes/ to edit the files.
>    - Any edits you make will automatically show up here. There's no need
>    to file a new PR for every edit.
>
> Be sure to *only* make edits that are relevant to this PR. So while
> you're on the revisingTgrainsDocfix branch, be sure only to edit
> SinOsc.schelp and TGrains.schelp.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/supercollider/supercollider/pull/2148#issuecomment-223641779>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG9xrgC3tbLM4NrkcJYbVfCZ0V56ST0qks5qIGSGgaJpZM4Itpl2>
> .
>



-- 
*T.* - Tiago Morais Morgado
http://tiagomoraismorgado.tk

**Comment by bagong** [comment]
Yea, sorry, from the results I realize how difficult it is to get into this, and how holey my explanation was. What you are used to seems clear, and yet in the explanation i forgot a lot of implicit knowledge.

Points I have gathered from this:

- it seems to be possible to accumulate changes in multiple files in a branch via the Web interface only. I didn't know that and it's nice.
- It is not only necessary to get clear that you want a branch for every PR, but also that you have to be very conscious about where you are when you create that branch. A branch destined to be a PR for master has to be created while on master (and not from the branch you were working on previously). So before creating a new branch you have to checkout master first.
- Oth you can keep pushing to a branch that is used for a PR, and the added commits will automatically appear in the PR. That is good, even required, if the commits belong to a single thought line. While we want thematically coherent PRs, we also want as few as possible to make the review process efficient.

**Comment by tiagmoraismorgado** [comment]
we are always learning a lot from each other. you were pretty, pretty generous with me today. and it's funny that sometimes in a learning-teaching exchange, you can learn a lot both from the side of observing the person you are actually tutoring. but also from the point of view of someone who is actually receiving knowledge. so exchange of knowledge, and community improvement. is  a constant. no matter if people know less or more about things. mostly because we are all different. no one is the same. and there are always means in which we can benefit from benefit int touch with each other

File: HelpSource/Classes/TGrains.schelp (hunk 1 above)
Comment (tiagmoraismorgado) [inline HelpSource/Classes/TGrains.schelp]: kislelk, i tried to fix this. i would appreciate if you could give me any further feedback on my corrections. let's move forward towards attempting to make my second contribution merging with the main supercollider repository, and who knows being part of a future release

File: HelpSource/Classes/TGrains.schelp [hunk 2]
```diff
@@ -58,138 +89,154 @@ Examples::
...
-	var trate, dur, clk, pos, pan;
-	trate = MouseY.kr(8,120,1);
-	dur = 12 / trate;
-	clk = Impulse.kr(trate);
-	pos = MouseX.kr(0,BufDur.kr(b)) + TRand.kr(0, 0.01, clk);
-	pan = WhiteNoise.kr(0.6);
-	TGrains.ar(2, clk, b, 1, pos, dur, pan, 0.1);
-}.scope(zoom: 4);
+SynthDef(\TgrainsExample2, {arg buffer, granular; var trate, dur, clk, pos, pan;
+	trate = MouseY.kr(8,120,1); dur = 12 / trate; clk = Impulse.kr(trate);
```
Comment (jamshark70) [inline HelpSource/Classes/TGrains.schelp]: For clarity, please keep each variable assignment on a separate line. Do not collapse multiple statements onto one line. In help files, we should err on the side of code readability.

File: HelpSource/Classes/TGrains.schelp [hunk 3]
```diff
@@ -3,12 +3,43 @@ summary:: Buffer granulator.
...
+
+At low speeds of playback, the result is a kind of soundscape, often described as a cloud, that is manipulatable in a manner unlike that for natural sound sampling or other synthesis techniques. At high speeds, the result is heard as a note or notes of a novel timbre. By varying the waveform, envelope, duration, spatial position, and density of the grains, many different sounds can be produced.
+
+Both have been used for musical purposes: as sound effects, raw material for further processing by other synthesis or digital signal processing effects, or as complete musical works in their own right. Conventional effects that can be achieved include amplitude modulation and time stretching. More experimentally, stereo or multichannel scattering, random reordering, disintegration and morphing are possible.
+
+from wikipedia the free encyplopedia
```
Comment (snappizz) [inline HelpSource/Classes/TGrains.schelp]: Can we make this a link instead of a copy-paste?

Comment (tiagmoraismorgado) [inline HelpSource/Classes/TGrains.schelp]: yes. i thought i had already done it

**Comment by telephon** [comment]
looks good. Is it ready to merge?

**Comment by tiagmoraismorgado** [comment]
telephon. i think so. maybe you should ask that in sc-development

**Comment by tiagmoraismorgado** [comment]
sorry telephone. i was now researching your work through an advice of bagging. and i realised that you are on of the leading developers in supercollider scene. if there is any feedback you would like to provide i would be really, really grateful

**Comment by telephon** [comment]
Could you please check through the other comments I gave you on the other helpfiles? Then you could perhaps check if your changes here could be improved in such a way. 

Also please could you please avoid exclamatory language in the help files? There is nobody who needs to be convinced how cool this all is. It is better if helpfiles are a bit dry.

**Pull request closed without merging**